src/Symfony/Component/ValueExporter/Formatter/CallableToStringFormatter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ValueExporter\Formatter;
+
+/**
+ * Returns a string representation of a string or array callable.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class CallableToStringFormatter implements StringFormatterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($value)
+    {
+        return is_callable($value) && !$value instanceof \Closure;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function formatToString($value)
+    {
+        if (is_array($value)) {
+            $caller = is_object($value[0]) ? get_class($value[0]) : $value[0];
+
+            if (isset($value[1])) {
+                $method = $value[1];
+
+                if (false !== $cut = strpos($method, $caller)) {
+                    $method = substr($method, $cut);
+                }
+
+                return sprintf('(callable) "%s::%s"', $caller, $method);
+            }
+
+            return sprintf('(invokable) "%s"', $caller);
+        }
+
+        return sprintf('(callable) "%s"', $value);
```

Comment:
This entire method could probably be reduced to have guard clauses instead. I think it will become a bit easier to read if so:
```php
public function formatToString($value)
{
    if (!is_array($value)) {
        return sprintf('(callable) "%s"', $value);
    }

    $caller = is_object($value[0]) ? get_class($value[0]) : $value[0];
    if (!isset($value[1])) {
        return sprintf('(invokable) "%s"', $caller);
    }

    $method = $value[1];
    if (false !== $cut = strpos($method, $caller)) {
        $method = substr($method, $cut);
    }

    return sprintf('(callable) "%s::%s"', $caller, $method);
}
```
*note that I didn't test it, just played around with it*